libs/wingsdk/src/core/app.ts

```diff
@@ -121,6 +123,12 @@ export class CdktfApp extends Construct implements IApp {
    * for unit testing.
    */
   public synth(): string {
+    // call preSynthesize() on every construct in the tree
+    if (!this.presynthed) {
+      preSynthesizeAllConstructs(this);
+      this.presynthed = true;
```

Comment:
Sure, I'll do that. (I added this because some tests were calling synth() twice, once in order to start the simulator, and a second time to obtain a snapshot, so alas...)